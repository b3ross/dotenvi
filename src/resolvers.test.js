const { resolvers } = require('./resolvers');

const awsSecretManagerResolver = resolvers.asm;

describe('AWS Resolver', () => {
  let consoleOutput = [];
  const mockedWarn = output => consoleOutput.push(output);
  beforeEach(() => (console.warn = mockedWarn));

  it('returns json secret value', () => {
    return awsSecretManagerResolver('jsonString.foo').then(output => {
      expect(output).toBe('bar');
    });
  });

  it('returns single string secret value', () => {
    return awsSecretManagerResolver('singleString').then(output => {
      expect(output).toBe('foobar');
    });
  });

  it('returns undefined if single string invalid', () => {
    return awsSecretManagerResolver('invalidSingleString').then(output => {
      expect(consoleOutput).toMatchSnapshot();
      expect(output).toBe(undefined);
    });
  });

  it('returns undefined if json string mapper invalid', () => {
    return awsSecretManagerResolver('jsonString.invalid').then(output => {
      expect(consoleOutput).toMatchSnapshot();
      expect(output).toBe(undefined);
    });
  });

  it('returns undefined if json key string invalid', () => {
    return awsSecretManagerResolver('invalidJsonString.invalid').then(output => {
      expect(consoleOutput).toMatchSnapshot();
      expect(output).toBe(undefined);
    });
  });
});
