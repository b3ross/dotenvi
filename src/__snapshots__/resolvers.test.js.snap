// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AWS Resolver returns undefined if json key string invalid 1`] = `
Array [
  "Secrets Manager can't find the specified secret.",
  "No aws secret manager value found for jsonString",
  "Secrets Manager can't find the specified secret.",
]
`;

exports[`AWS Resolver returns undefined if json string mapper invalid 1`] = `
Array [
  "Secrets Manager can't find the specified secret.",
  "No aws secret manager value found for jsonString",
]
`;

exports[`AWS Resolver returns undefined if single string invalid 1`] = `
Array [
  "Secrets Manager can't find the specified secret.",
]
`;
